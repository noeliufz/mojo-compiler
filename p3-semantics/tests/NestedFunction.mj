proc putchar(c: int): int;

{
  var i = 4;
  proc foo(): int {
    return i+1;
  }
  if foo() == 5 then { putchar(Ord('T')); }
  else { putchar(Ord('F')); }
  putchar(Ord('\n'));
}
