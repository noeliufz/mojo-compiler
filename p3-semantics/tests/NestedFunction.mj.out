BEGIN _BUILTIN
_BUILTIN.int=int
_BUILTIN.bool={false:bool=0, true:bool=1}
_BUILTIN.false:bool=0
_BUILTIN.true:bool=1
_BUILTIN.byte=[-128..127]
_BUILTIN.char={}
_BUILTIN.Null=ref null
_BUILTIN.null:Null=0
_BUILTIN.Refany=ref null
_BUILTIN.Address=ref null
_BUILTIN.Root=null object {}
_BUILTIN.Text=ref null
_BUILTIN.First:(...)
_BUILTIN.Last:(...)
_BUILTIN.Ord:(...)
_BUILTIN.Val:(...)
_BUILTIN.New:(...)
_BUILTIN.Number:(...)
_BUILTIN.Loophole:(...)
BEGIN NestedFunction
NestedFunction.putchar:(VALUE c:int):int
BEGIN NestedFunction.2
NestedFunction.2.i:int
NestedFunction.2.foo:():int
BEGIN NestedFunction.2.foo
END NestedFunction.2.foo
END NestedFunction.2
END NestedFunction
END _BUILTIN
